packages/core/src/lib/base_protocol.ts

```diff
@@ -60,4 +65,63 @@ export class BaseProtocol implements IBaseProtocol {
     );
     return peer;
   }
+
+  /**
+   * Retrieves a list of peers based on the specified criteria.
+   *
+   * @param numPeers - The total number of peers to retrieve. If 0, all peers are returned.
+   * @param maxBootstrapPeers - The maximum number of bootstrap peers to retrieve.
+   * @returns A Promise that resolves to an array of peers based on the specified criteria.
+   */
+  protected async getPeers({
+    numPeers,
+    maxBootstrapPeers
+  }: {
+    numPeers: number;
+    maxBootstrapPeers: number;
+  }): Promise<Peer[]> {
+    // Retrieve all peers that support the protocol
+    const allPeersForProtocol = await getPeersForProtocol(this.peerStore, [
+      this.multicodec
+    ]);
+
+    // Collect the bootstrap peers up to the specified maximum
+    const bootstrapPeers = allPeersForProtocol
+      .filter((peer) => peer.tags.has(Tags.BOOTSTRAP))
+      .slice(0, maxBootstrapPeers);
+
+    // Collect non-bootstrap peers
+    const remainingPeers = allPeersForProtocol.filter(
+      (peer) => !bootstrapPeers.includes(peer)
```

Comment:
this is incorrect as you are filtering out peers that you previously selected. but if you selected 1 out of 3 bootstrap peers then `remainingPeer` will contain 2 bootstrap peers